### PR TITLE
Add useDirection React hook

### PR DESCRIPTION
**Pull request by micmro**
I have created this React hook that return the currently active `Direction` and re-emits when it changes that I thought might be useful for other users of this extension, e.g. when requiring the direction for configuring additional state that would usually be set globally based on API responses.

Can be imported from `@pxblue/storybook-rtl-addon/useDirection`

#### Changes proposed in this Pull Request:
- add `useDirection` hook
- export it via `@pxblue/storybook-rtl-addon/useDirection`
- rename `types.d.ts` to `types.ts` to include the types as part of the compile-target in `dist`

#### Screenshots:
used in decorator
```tsx
import useDirection from "@pxblue/storybook-rtl-addon/useDirection";

export const decorators = [
  (Story) => {
    const direction = useDirection();

    return (
      <>
        {direction}
        <hr />
        <Story />
      </>
    );
  },
];

```
Screen Recording 2021-10-03 at 10 06 44 PM

#### To Test:
I've setup an example repo (https://github.com/micmro/react-storybook-storybook-rtl-addon-test) you can clone and test via the steps documented in https://github.com/pxblue/storybook-addons/blob/dev/rtl/README.md#local-testing

### CODE REVIEW

**Review by emclaug2 — COMMENTED** [review]
Thanks for the PR! I'm taking a look at this now.  I ran your example project & I understand the issue that you were facing with the original `getDirection` export.  

A couple notes:
- Your example project uses a JS storybook & ours uses TS.  As the hook is right now, it's missing some type definitions needed to run in a TS environment; adding it to our storybook throws an error.

image

I managed to fix this issue by including: 
`export * from "./useDirection";` in the index.tsx file.


- I'm thinking to keep the `useDirection` import in line with how `getDirection` has been imported:
```
import { useDirection } from '@pxblue/storybook-rtl-addon';
```

- We'll need some documented differentiation between `useDirection` and `getDirection`.   I don't want this to be a breaking change for `getDirection` so renaming `useDirection` to something like `getDirectionHook` or something of the sort to help differentiate would be good.

**Comment by emclaug2** [comment]
After this gets merged in, this would be a good time to introduce some Angular vs React example docs as well.  #9 in our issues list looks relevant.

**Comment by micmro** [comment]
Thanks for the review @emclaug2. Will look into it this week (a bit busy week, so it might take a few days).

**Comment by micmro** [comment]
Hi @emclaug2, I've setup a TS example storybook https://github.com/micmro/-react-ts-storybook-rtl-addon-test - now the types for `@pxblue/storybook-rtl-addon/useDirection` get work correctly:

Screen Shot 2021-10-07 at 8 41 14 PM

> renaming useDirection to something like getDirectionHook or something of the sort to help differentiate would be good.

The `use` prefix is required for hooks (see [React docs](https://reactjs.org/docs/hooks-custom.html#extracting-a-custom-hook)), so it could be renamed to `useDirectionHook` - from a React perspective that would be redundant, but might make it easier to understand for other frameworks.

I thought this could also be addressed by keeping `@pxblue/storybook-rtl-addon/useDirection` a separate entry file, so there would also not be any potential for bundling issues. While Storybook uses React anyway, there can be some storybook setups that rewrite some type paths etc and it would be less desirable to have to re-map react.

File: rtl/package.json [hunk 1]
```diff
@@ -9,7 +9,9 @@
     "dist",
     "README.md",
     "LICENSE",
-    "register.js"
+    "register.js",
+    "useDirection.js",
+    "useDirection.d.ts"
```
Comment (micmro) [inline rtl/package.json:14]: I've setup a separate types-entry file for `@pxblue/storybook-rtl-addon/useDirection` to keep the index file free of React components.

**Comment by emclaug2** [comment]
Thanks for the updates here @micmro!  I'll take a look at this first thing tomorrow.

**Comment by emclaug2** [comment]
The code all checks out within our React typescript project as well.  I'm thinking this is good to merge. 

I'll make a follow-up PR with a new CHANGELOG entry & an updated integration section in the README.  #9 

Thanks for your contribution @micmro!  Mind if I tag you in the follow-up PR just so the new usage-section reads correctly?

**Review by emclaug2 — APPROVED** [review]

**Comment by micmro** [comment]
Thanks @emclaug2 . Sure, feel free to tag me. Let me know if you'd like me to help out with some examples.

**Pull request merged**